docs/spec_draft.md

```diff
@@ -2279,6 +2280,85 @@ where `i[d] = j[permutation[d]]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.triangular_solve
+
+### Semantics
+
+Solves systems of linear equations of a batch of matrices with lower or upper
+triangular coefficients by forward- or back-substitution.
+
+More formally, given `a` and `b`, `result[i0, ..., iR-3, :, :]` is the solution
+to `op(a[i0, ..., iR-3, :, :]) * x = b[i0, ..., iR-3, :, :]` when `left_side` is
+`true` or `x * op(a[i0, ..., iR-3, :, :]) = b[i0, ..., iR-3, :, :]` when
+`left_side` is `false`, solving for the variable `x` where `op(a)` is either
+`op(a) = a`, or `op(a) = Transpose(a)`, or `op(a) = Conj(Transpose(a))`.
+
+Input data is read only from the lower triangle of `a`, if `lower` is `true` or
+upper triangle of `a`, otherwise. Output data is returned in the same triangle;
+the values in the other triangle are implementation-defined.
+
+If `unit_diagonal` is true, the diagonal elements of `a` are assumed to be 1 and
+not accessed.
```

Comment:
Proposed reformulation:

```
If `unit_diagonal` is true, then the implementation can assume that the diagonal
elements of `a` are equal to 1, otherwise the behavior is undefined.
```

Feedback:
  1) "the implementation can assume" is consistent with other similar ops.
  2) "and not accessed" is dropped to let the implementations decide on their 
     own what they want to do with this knowledge.
  3) "otherwise the behavior is undefined" needs to be said explicitly.